mock/py/mockbuild/mounts.py

```diff
@@ -85,17 +85,30 @@ def __repr__(self):
 class BindMountPoint(MountPoint):
     '''class for managing bind-mounts in the chroot'''
     @traceLog()
-    def __init__(self, srcpath, bindpath):
+    def __init__(self, srcpath, bindpath, recursive=False, options=None):
         MountPoint.__init__(self, mountsource=srcpath, mountpath=bindpath)
         self.srcpath = srcpath
         self.bindpath = bindpath
+        self.recursive = recursive
+        self.options = options
         self.mounted = self.ismounted()
 
     @traceLog()
     def mount(self):
         if not self.mounted:
-            util.mkdirIfAbsent(self.bindpath)
-            cmd = ['/bin/mount', '-n', '--bind', self.srcpath, self.bindpath]
+            if os.path.isdir(self.srcpath):
+                util.mkdirIfAbsent(self.bindpath)
+            elif not os.path.exists(self.bindpath):
+                util.touch(self.bindpath)
+            cmd = ['/bin/mount', '-n']
+            if self.recursive:
+                cmd.append('--rbind')
+            else:
+                cmd.append('--bind')
+            bind = '--rbind' if self.recursive else '--bind'
```

Comment:
This variable is not used at all. Probably some leftover.

Reply:
Yep, removed.